src/Components/ABDM/LinkABHANumberModal.tsx

```diff
@@ -159,11 +171,13 @@ const ScanABHAQRSection = ({
         value={qrValue}
         disabled={!!authMethods.length}
         onChange={(value) => {
-          if ([2, 7, 12].includes(value.length)) {
-            if (qrValue.length && qrValue[qrValue.length - 1] === "-") {
-              value.slice(value.length - 1);
-            } else {
-              value += "-";
+          if (value[0] && !isNaN(Number(value[0]))) {
+            if ([2, 7, 12].includes(value.length)) {
```

Comment:
Can we add a small comment defining the purpose of these 3 magic numbers? `2, 7, 12`